Extensions/CompositeMetadata.md

```diff
@@ -0,0 +1,45 @@
+# Composite Metadata Extension
+
+_This extension specification is currently incubating._
+
+## Introduction
+There are a number of situations where an arbitrary collection of discrete metadata types should be attached to frame.  For example, a request frame may want to include both routing metadata as well as tracing metadata.  This extension specification provides an interoperable structure for metadadata payloads to contain multiple discrete metadata types.  It is designed such that if a consumer of the metadata is unaware of a particular type, it can be safely skipped and the next one read.
```

Comment:
I wonder whether safely skipping an unknown header should be a shared flag?  Should it be possible to require a responder to understand the metadata to proceed?

Reply:
Maybe? Like if you sent a frame with routing data you could for it throw an exception if it didn’t understand?

Reply:
That isn't a required semantic for any other use of metadata in the spec though.  If you put a metadata MIME-type in ` SETUP` frame, there's no _mandate_ that the server fail if it isn't understood.